### ISSUE TITLE
SSTU Modifications for RO

### ISSUE THREAD

**Issue by pap1723** [KSP-RO/RealismOverhaul#1625]
I will post about issues I have found and what needs to be / should be fixed. I will include some check boxes so that I can show you what I have completed.

**Apollo**
- [x] The AJ10-137 integrated into the Apollo SM did not have a plume, fixed using the same data from the AJ10-137 standalone engine
- [x] Fuel Cell and Ox converter were not creating enough electricity and Oxygen, I used the information from FASA and copied it to the Apollo CM

**SSTU Procedural Tanks / Parts**
- [x] There is a TECHLIMITS file in the RP-0 folder in order to unlock the correct diameters with the proper nodes. This is deprecated and now uses the PARTUPGRADE system new in 1.2.2. These PARTUPGRADES need to be placed in the proper nodes to set the proper max diameter of the procedural parts and tanks in SSTU

**Global Test Flight**
- [x] AJ10-137 did not have any Test Flight settings, I created them
- [x] F-1 did not have any Test Flight settings, I created them (still needs to be checked)

**Comment by assassinacc** [comment]
I've started with the tech limits https://github.com/KSP-RO/RealismOverhaul/compare/master...assassinacc:SSTU_PartUpgrades

**Comment by Vanya-Kerbalnovich** [comment]
Fwiw SSTU's modular SRBs show zero thrust and a burntime of NaN on the VAB. I can't code to save my life (learning though!), otherwise I'd gladly help.

**Comment by blowfishpro** [comment]
Going to note issues here as I find them, may work on some later:

* 5x F-1 is still showing up despite what this issue says, not sure why
* 5x SSME is a part but does not need to be
* Procedural decoupler has major diameter increments set to 0.1/0.2m currently, should be 0.5/1m to be consistent with everything else
* Upper stage tank is currently marked non-RO.  This might be hard to get working as RF tanks can't currently support RCS being highly pressurized and main tank not being
* A couple of modular SRBs still show up and do not work
* Non-modular SRBs based on SSTU models are gray probably based on recoloring features that are still in SSTU's dev branch but will be released soon

**Comment by blowfishpro** [comment]
* RD-171/180/181 need RO configs

**Comment by blowfishpro** [comment]
* [ ] Orion service module cannot jettison its side panels and they stay attached forever

**Comment by blowfishpro** [comment]
* [ ] Orion LES could possibly use a bit more thrust in jettison mode - it can't clear the rocket while under a decent amount of acceleration.  I don't know if the current number is based on a specific reference.

**Comment by Theysen** [comment]
Is it supposed to be jettisoned during thrust phases? Every render I saw this far had the LES jettison at main stage separation before the upper stage ignited. Might be due to video and entertainment purposes of course.

**Comment by pap1723** [comment]
- [ ] The Modular Heat Shield is not working correctly. It allows you to resize, but a 20m? (very large) model is always showing

**Comment by blowfishpro** [comment]
@Theysen you could be right, I haven't looked at the animations

**Comment by c-mons** [comment]
> Orion service module cannot jettison its side panels and they stay attached forever

@blowfishpro Jettison side panels of the Orion European Service Module worked fine for me when I tried it (SSTU-0.5.34.134)

**Comment by blowfishpro** [comment]
I was on the dev branch.  That might be the difference.  I'll check it in vanilla SSTU as well, as the problem might be there.

**Comment by shadowmage45** [comment]
Just thought I would add -- if anyone needs any plugin-side changes made to SSTU to make the RO fixes possible, please let me know (PRs welcome as well, and preferred if possible).

(I realize this is a long-running / older issue ticket, but many of the problems still seem relevant)

**Comment by Theysen** [comment]
@shadowmage45 As of right now, the actual big thing holding SSTU for RO back is 1.2.2 and not being on 1.3. I have no idea if that would even work and can be achieved - I highly doubt it - but a backport of 1.3 plugins to 1.2.2 so we can use all those new features like recoloring etc is out of scope right? 

Currently we have no ETA for a version bump for 1.3 and SSTU keeps on drooling with new features.

Don't consider this as any request actually or something else, it is just a question but I'd assume I can give myself the answer. 


As for any other changes pluginside, I'd actually just could think of making the AutoDecouple feature a procedural where the input timer can be set by the user.

**Comment by blowfishpro** [comment]
1.3 RO should be relatively close, I think there were only a few things we were waiting on

**Comment by PhineasFreak** [comment]
IIRC the RO team was awaiting the 1.3.1 release to happen, since the localization implementation on KSP 1.3.0 affected the overall performance. It is also one of the major bullet points in the changelog.

**Comment by shadowmage45** [comment]
@Theysen -- a backport would be fairly labor intensive, mostly due to having to recompile the shader packs, as well as fix the actual plugin code;  not something that I will be doing, however I wouldn't be opposed if someone else wanted to undertake the task.

Auto-decouple timing slider -- sure:
https://github.com/shadowmage45/SSTULabs/issues/578

(will probably be available with the next release)

**Comment by Starwaster** [comment]
Has anyone looked at the mass and thrust? For the ones I've compared (J-2, J-2X, F-1 and F-1B) they typically are 50% more massive and have 40% of the thrust of their real life counterparts. (those are approximate figures, it varies from part to part) 

I can't find anywhere in the repository where these are being adjusted (some plume configs and parts resized but that's it)

**Comment by Theysen** [comment]
They're all using the enginetype presets, I suppose there is a hickup with the modules of SSTU and the way RO Global configs patch the engines. I heard from multiple users that engine masses were wrong once clustering happens via the SSTU slider. 

I actually also don't know how feasible it is to fix this before going to 1.3.1 because SSTU changed A LOT since 1.2.2 compatability. I guess a complete overhaul is needed?

**Comment by Starwaster** [comment]
@Theysen so you're saying there is or was a means of scaling the thrust and mass for these engines aside from directly setting the mass (in the part) and the thrust (in the engine module)?

Can you elaborate on that and point me at the relevant files?

(FYI they definitely do respond to setting part mass and engine maxThrust as one would expect and it works with clustering so it's definitely feasible, just needs the 'elbow grease ' so to speak)

**Comment by Starwaster** [comment]
Ok, after scouring the hell out of the repository - ALL branches, I finally found a branch (e1_engine) where some engines had mass adjusted and some had thrust adjusted, or both. But either it never merged with  either master or main or the changes got reverted..

**Comment by JPROSS87** [comment]
I've been running the dev 1.31 RO + RP0 for a few days and I can't replicate the thrust issue with the SSTU engines you describe - the thrust seems to match up exactly with the non-SSTU versions, including clustered versions. I do see the mass problem though. Specifically, a SSTU cluster is (2n-1) times as massive as it should be, so the single engines are fine but a cluster of 4 has the same mass as 7 equivalent engines etc. I can't see how the bug occurs but perhaps this will offer a clue to someone else. 

As for the rest of SSTU everything is working fine for me except:

1) The two variants of the upper stage tank have since been merged into a single part, so the realfuels, RO and RP0 configs needed to be edited for those specific tanks, ie SSTU-SC-TANK-MUS-* should now be SSTU-SC-TANK-MUS.

2) The upper stage avionics can still control infinite mass. There is a comment in the config that this had been addressed but that may have been prior to the new avionics stuff.

**Comment by shadowmage45** [comment]
Re: Engine-mass issues --
This line in the engine module (and configs) should control whether the SSTU plugin/module does any mass adjustments
https://github.com/shadowmage45/SSTULabs/blob/master/Plugin/SSTUTools/SSTUTools/Module/SSTUModularEngineCluster.cs#L124

When set to false, all mass adjustments should be handled by ModuleEngineConfigs.  If this is not the case, please let me know and I can investigate further.

Could also be the case that I'm not updating the ModuleEngineConfig properly on cluster # changes, but I'm far from an expert on RF / MEC setup or use;  if there is something that needs to be changed in this method to clean up compatibility, please let me know --
https://github.com/shadowmage45/SSTULabs/blob/master/Plugin/SSTUTools/SSTUTools/Util/SSTUModInterop.cs#L34-L53

**Comment by winged7** [comment]
I've noticed that solar paneles used by Orion are about 4x as apowerful as they should be, so something is wrong here:

`@MODULE[SSTUSolarPanelDeployable]
	// http://www.esa.int/Our_Activities/Human_Spaceflight/Orion/Power
	{
		@resourceAmount = 11.2
	}`

https://github.com/KSP-RO/RealismOverhaul/blob/master/GameData/RealismOverhaul/RO_SuggestedMods/SSTU/SSTU_Orion.cfg

**Comment by pap1723** [comment]
OLD
